src/generators/javascript/templates/index.hbs

```diff
@@ -327,6 +327,126 @@ export function {{functionName}}(
 {{/if}}
 {{/each}}
 
+{{#each page}}
+{{> functionDocumentation isBrowser=../isBrowser }}
+{{#if ../isBrowser}}
+export function {{functionName}}
+(
+	props{{#if isPropertiesOptional}}?{{/if}}: {{propertiesType}},
+	options?: ApiOptions,
+	callback?: apiCallback
+): void {
+	const a = analytics()
+	{{#if ../isDevelopment}}
+	const schema = {{rawJSONSchema}}
+	const message = {
+		properties: props || {},
+		options,
+	}
+	validateAgainstSchema(message, schema).then(()=>{
+		if (a) {
+			a.page(
+				props || {},
+				withRudderTyperContext(options),
+				callback
+			)
+		}
+	})
+	{{else}}
+	if (a) {
+		a.page(
+			props || {},
+			withRudderTyperContext(options),
+			callback
+		)
+	}
+	{{/if}}
+}
+{{else}}
+{{!-- TODO for Node--}}
+{{/if}}
+{{/each}}
+
+{{#each identify}}
+{{> functionDocumentation isBrowser=../isBrowser }}
+{{#if ../isBrowser}}
+export function {{functionName}}
+(
+	props{{#if isPropertiesOptional}}?{{/if}}: {{propertiesType}},
+	options?: ApiOptions,
+	callback?: apiCallback
+): void {
+	const a = analytics()
+	{{#if ../isDevelopment}}
+	const schema = {{rawJSONSchema}}
+	const message = {
+		traits: props || {},
+		options,
+	}
+	validateAgainstSchema(message, schema).then(()=>{
+		if (a) {
+			a.identify(
+				props || {},
+				withRudderTyperContext(options),
+				callback
+			)
+		}
+	})
+	{{else}}
+	if (a) {
+		a.identify(
+			props || {},
+			withRudderTyperContext(options),
+			callback
+		)
+	}
+	{{/if}}
+}
+{{else}}
+{{!-- TODO for Node--}}
+{{/if}}
+{{/each}}
+
+{{#each group}}
+{{> functionDocumentation isBrowser=../isBrowser }}
+{{#if ../isBrowser}}
+export function {{functionName}}
+(
+	props{{#if isPropertiesOptional}}?{{/if}}: {{propertiesType}},
```

Comment:
For Group and Identify APIs, there is no concept of properties. It is traits.
It's fine if we want to treat all those as properties internally but in the exposed API to the users, the interface should be identical to the SDKs.

Reply:
In the argument list of the API we are passing it as traits

Reply:
In the rudder typer interface to the user, are they named as properties or traits? It should be traits. Otherwise, it is confusing.